bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java

```diff
@@ -224,35 +230,41 @@ protected void doConnectedRun() throws IOException, ApiException {
                 lights = hueBridge.getFullConfig().getLights();
             }
 
+            final LightStatusListener discovery = discoveryListener;
```

Comment:
Please keep the different states (added, removed and changed) for each kind of listener, without assuming what methods will be used by each listener.

Reply:
I'm already thinking about doing even more refactoring, because in my opinion the discovery service should not have the same Interface like light and sensor handler to make clear which events are really used by the single thing. The discovery service needs only add(Light|Sensor)Discovery and remove(Light|Sensor)Discovery.

Reply:
Yes that is an option I was thinking of. Two different event listeners.

Reply:
I have done a refactoring of discovery service. Please have a look.